src/ngraph/pass/manager.cpp

```diff
@@ -122,6 +131,19 @@ void ngraph::pass::Manager::run_passes(shared_ptr<Function> func)
             }
         }
         index++;
+        pass_timer.stop();
+        if (profile_enabled)
+        {
+            PassBase* p = pass.get();
+            string name = typeid(*p).name();
+            int status;
+            name = abi::__cxa_demangle(name.c_str(), 0, 0, &status);
```

Comment:
This is so awesome!! I didn't realize we can call a demangler programmatically.

Reply:
My goal is to blow your mind